src/System.Runtime.Environment/src/OSName.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSName : IEquatable<OSName>
+    {
+        private readonly string _name;
+
+        private const string WindowsName = "WINDOWS";
+        private const string LinuxName = "LINUX";
+        private const string OSXName = "OSX";
+
+        public OSName(string name)
+        {
+            _name = name;
+        }
+
+        public static OSName Windows { get { return new OSName(WindowsName); } }
+        public static OSName Linux { get { return new OSName(LinuxName); } }
+        public static OSName OSX { get { return new OSName(OSXName); } }
```

Comment:
Apple just calls it OS X nowadays, quote from [Wikipedia](http://en.wikipedia.org/wiki/OS_X#History):

> In 2012, with the release of OS X 10.8, the "Mac" prefix was officially dropped in all references to the operating system name within its web site.